### DISCUSSION TITLE
How to disable the context menu when clicking with Ctrl modifier?

### DISCUSSION THREAD

**Discussion by huyz** [macvim-dev/macvim#1324] [Q&A] [answered]
When I hold Ctrl and click in the window, a context menu appears.
How can I disable it?

It conflicts with https://github.com/mg979/vim-visual-multi/wiki/Mappings#mouse-mappings

**Comment by eirnym** [comment]
Ctrl-Click is a default key combination for macOS to trigger context menu in any application.

is it possible to change these key mappings to something like Ctrl-Shift-click?

**Reply by huyz** [reply]
I would rather disable the context menu altogether as I don't find any of the menu entries useful at all as a mainly keyboard user (while the mouse is quite useful for multiple cursors).

Take for example VimR: no context menu, no conflict

**Comment by ychin** [comment, marked as answer]
Do the following in Terminal:

```
defaults write org.vim.MacVim MMTranslateCtrlClick 0
```

There isn't a user-visible setting for this in the Settings pane, but I may add one in the "Input" category to make it more discoverable (Edit: See #1326). You can do `:help MMTranslateCtrlClick` in MacVim to read more about the setting. But it will basically do what you want.

**Reply by ychin** [reply]
On a tangent: A more Mac-like shortcut would be to map `<D-LeftMouse>` instead as Cmd keys should be used for things like this. *That said*, seems like that doesn't work though as doing a Cmd-click would just result in `<LeftMouse>`. I filed https://github.com/macvim-dev/macvim/issues/1325 to fix this.

**Reply by huyz** [reply]
That makes sense.  As soon as #1325 I'll switch from `<C-LeftMouse>` to `<D-LeftMouse>`

**Reply by huyz** [reply]
@ychin What about <C-S-LeftMouse>?  I still can't seem to get that reported.

**Reply by ychin** [reply]
Do you mean `<D-LeftMouse>`? I think the issue is still open? It's not going to be done for a while. I have other higher priority things to get done.

**Reply by huyz** [reply]
Oh oops. I forgot to escape the HTML-tag looking thing and it didn't render. 
I meant `<C-S-LeftMouse>`

**Reply by ychin** [reply]
Oh, `<C-S-LeftMouse>` that does seem to not be working. I can take a quick look to see if it's a quick fix or have to be fixed along `<D-LeftMouse>` at a later date.

**Reply by ychin** [reply]
Actually I think Vim is hard-coded to use Ctrl-Shift-leftmouse to do modeless selection, so it's not specific to MacVim. See `:h modeless-selection`:

> A different kind of selection is used when:
> - in Command-line mode
> - in the Command-line window and pointing in another window
> - at the |hit-enter| prompt
> - whenever the current mode is not in the 'mouse' option
> - when holding the CTRL and SHIFT keys in the GUI

**Reply by huyz** [reply]
Interesting. But isn't selecting different from clicking? I figure `LeftClick` and `LeftDrag` can be handled differently.

**Reply by ychin** [reply]
Yeah but that's just how Vim handles it today for simplicity.
